### PR TITLE
Add miracl fr reranking dataset

### PR DESCRIPTION
**Pull request by shreeya-dhakal**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Reranking/fra/MIRACLFrReranking.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        domains=None,
+        task_subtypes=None,
+        license="Not specified",
+        socioeconomic_status="high",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation=None,
+        n_samples={"test": 343},
+        avg_character_length=None,
```
Comment (asparius) [inline mteb/tasks/Reranking/fra/MIRACLFrReranking.py:32]: This is required!

File: mteb/tasks/Reranking/fra/MIRACLFrReranking.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        date=("2023-11-25", "2023-11-27"),  # using dates on commits for date
+        form=["written"],
+        domains=None,
+        task_subtypes=None,
+        license="Not specified",
+        socioeconomic_status="high",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation=None,
```
Comment (asparius) [inline mteb/tasks/Reranking/fra/MIRACLFrReranking.py:30]: Add MIRACL paper citation

File: mteb/tasks/Reranking/fra/MIRACLFrReranking.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            "revision": "50bd0db5fa3d5845142fd58d55f870b5bbb27960",
+        },
+        type="Reranking",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["fra-Latn"],
+        main_score="map",
+        date=("2023-11-25", "2023-11-27"),  # using dates on commits for date
+        form=["written"],
+        domains=None,
```
Comment (asparius) [inline mteb/tasks/Reranking/fra/MIRACLFrReranking.py:23]: It is collected from Wikipedia, so web and Encyclopaedic

**Comment by imenelydiaker** [comment]
I think that a task integrating MIRACL with all languages is coming, see these issues: https://github.com/embeddings-benchmark/mteb/issues/198 and https://github.com/embeddings-benchmark/mteb/issues/354

**Comment by KennethEnevoldsen** [comment]
Will close this following @imenelydiaker comment, thanks for the contribution @shreeya-dhakal. You might consider examining the open issues to find alternative datasets to add

**Pull request closed without merging**